storage/storage.go

```diff
@@ -0,0 +1,32 @@
+package storage
+
+import internal "github.com/containers/image/v5/internal/storage"
+
+// Note that the storage package is merely aliasing *some* parts of the
+// `internal/storage` package.  This allows us to control the public parts of
+// the API (i.e., in this file) and the c/image internal parts of the API
+// (i.e., unaliased parts).  An immediate use case is to add new features
+// (e.g., to the `copy` package) without having to worry too much about public
+// facing changes and versioning.
```

Comment:
BTW I don’t think this is actually true. Once a camel-case method exists, it can be called by anyone using a matching interface, whether or not the implementor is in a visible package or somewhere in `internal`.

So there isn’t much difference between
- Private interface in `c/internal/storage`; implementation in `c/image/storage` as a private type, returns `types.imageDestination`
- Private implementation in `c/internal/storage` as an exported type, returns `types.ImageDestination`

The caller is going to make a very similar check, and external users can do the same things. So I don’t think this move + forwarding is _necessary_.

OTOH a major difference is that once this is structured as a method on a type, not an interface, it’s impossible for any other internal transport to directly adopt the new API without extracting the interface.

Reply:
I think I overstretched the statement.  If it's interface you're right on point. In this case, the extensions to `StorageImageDestination` are methods outside of the API, so I don't think any outside callers could access the `*WithOptions` methods.  Do you agree?

Reply:
I got the syntax wrong earlier (a simple cast does statically verify that the method is known to exists), but AFAIK a dynamic check like this always works:
```go
package main

import (
	"fmt"
	"io"
)

type myType struct{}

func (myType) Close() error       { return nil }
func (myType) NotInAnyInterface() { fmt.Println("Hello!") }

func constructor() io.Closer { return &myType{} }

func writeToCloser(closer io.Closer) {
	type bypass interface{ NotInAnyInterface() }
	bypassed := closer.(bypass)
	bypassed.NotInAnyInterface()
}

func main() {
	writeToCloser(constructor())
}
```